src/components/journey/Journey.js

```diff
@@ -1,15 +1,180 @@
-import React from 'react'
-import './Journey.css'
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom'
+import JourneyCard from './../journey-card/JourneyCard'
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import Avatar from '@material-ui/core/Avatar';
+import Grid from '@material-ui/core/Grid';
+import Divider from '@material-ui/core/Divider';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
 
-// Component used to display a journey
-const Journey = ({ destination, price }) => (
-    <div>
-        <div>
-            <h1>Destination: {destination}</h1>
-            <p>Price: {price}</p>
-        </div>
-    </div>
-)
+const styles = theme => ({
+    container: {
+        flexDirection: 'column',
+    },
+    bigAvatar: {
+        margin: 10,
+        width: 250,
+        height: 250,
+    },
+    root: {
+        width: '100%',
+        maxWidth: 360,
+        backgroundColor: theme.palette.background.paper,
+        width: 500,
+        margin: '0 auto',
+    },
+    chip: {
+        marginRight: theme.spacing.unit,
+    },
+    section1: {
+        margin: `${theme.spacing.unit * 3}px ${theme.spacing.unit * 2}px`,
+    },
+    section2: {
+        margin: theme.spacing.unit * 2,
+    },
+    section3: {
+        margin: `${theme.spacing.unit * 6}px ${theme.spacing.unit * 2}px ${theme.spacing.unit * 2}px`,
+    },
+});
 
-export default Journey;
+class Journey extends Component {
 
+    constructor(props) {
+        super(props);
+        console.log('hello : ' + props.match.params.id);
+        // Contains the state of the component
+        this.state = {
+            error: null,
+            isLoaded: false,
+            journey: [],
+            journeyId: props.match.params.id
+        }
+
+        console.log('before : ' + props.match.params.id);
+        // It allows for the function to get "this"
+        this.findJourney = this.findJourney.bind(this);
+        console.log('hello : ' + props.match.params.id);
+    };
+
+    componentDidMount() {
+        this.findJourney();
+    }
+
+    /**
+     * Bring a journey from the server (graphql)
+     */
+    getJourney = async (query, variables) => {
```

Comment:
please try to use rxjs instead of async, see if that's possible

Reply:
Ok I'll check

Reply:
This part is not useful because we did not really search by id, as a user.
But I look on the internet, and I did not find something interesting with rxjs for react.

Reply:
I made a mistake, we use it in fact, I'll change that to do like the others request.

Reply:
In fact I saw that I corrected in on the other branch. I merged it, I think it's better.